source/MissionAction.cpp

```diff
@@ -419,9 +426,13 @@ void MissionAction::Do(PlayerInfo &player, UI *ui, const System *destination, co
 			subs["<ship>"] = player.Flagship()->Name();
 		string text = Format::Replace(dialogText, subs);
 		
+		// Don't push the dialog text if this is a visit action; on visit dialogs 
+		// are handled by PlayerInfo as to avoid the player being spammed by 
+		// dialogs if they have multiple missions active with the same destination
+		// (e.g. in the case of stacking bounty jobs).
 		if(isOffer)
 			ui->Push(new Dialog(text, player, destination));
-		else
+		else if(trigger != "visit")
```

Comment:
I think this feature is great for avoidance of spamming from minor jobs. But if a major storyline job (or a job with a deadline) has an "on visit", then I would like to see that one regardless of how many other jobs are relevant for the current location.

Reply:
How about only condensing the on visit dialogs from missions that return false for `IsUnique()`, otherwise have it shown here? That would distinguish between jobs (plus repeatable spaceport missions) and proper missions. Then in terms of deadlines, since the player might have multiple deadline based jobs, how about just adding another line that says something along the lines of "X of your unfinished missions here are time-sensitive/have an approaching deadline," instead of just showing all missions with deadlines?

Reply:
I prefer my suggestion of stacking messages with the same mission `Identifier` (e.g. all `"Bounty Hunting (Small)"`), since it makes sure the player still gets reminded of the various types of conditions they haven't satisfied, even though it still results in showing a handful of visit dialogs. At minimum, I agree that non-repeated missions should not be treated as though they are spammy since they are, by definition, not.

Another recommendation from #4565 would be to add a mechanism that prevents displaying a mission's "on visit" dialog more than once per session (that could be done by adding a non-serialized boolean to the MissionAction class).

Reply:
![image](https://user-images.githubusercontent.com/17688683/66705193-b9345980-ecf1-11e9-9818-787737496b82.png)